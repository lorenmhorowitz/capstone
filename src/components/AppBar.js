import AppBars from "@material-ui/core/AppBar";
import { createMuiTheme, IconButton } from "@material-ui/core";
import { Link } from "react-router-dom";
import MenuIcon from "@material-ui/icons/Menu";
import MuiThemeProvider from "@material-ui/core/styles/MuiThemeProvider";
import PropTypes from "prop-types";
import React, { Component } from "react";
import styles from "../css/Component.css";
import Toolbar from "@material-ui/core/Toolbar";
import { withStyles } from "@material-ui/core/styles";

const muiTheme = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  palette: {
    primary: {
      light: "#757ce8",
      main: "#4D4847",
      dark: "#f78a4d",
      contrastText: "#000000"
    },
    secondary: {
      main: "#FFFFFF"
    }
  }
});

class AppBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      anchorEl: null,
      redirectHome: false
    };
  }

  render() {
    const { anchorEl } = this.state;
    const open = Boolean(anchorEl);
    return (
      <div>
        <MuiThemeProvider theme={muiTheme}>
          <AppBars id="root" color="primary">
            <Toolbar>
              <Link to="/home" id="Logo">
                <span id="HomeTitle">Corner</span>
                <span id="StoneTitle">Stone</span>
              </Link>

              <div id="menuButton">
                <IconButton
                  aria-owns={open ? "Menu" : null}
                  aria-haspopup="true"
                  onClick={this.handleMenu}
                  aria-label="Menu"
                  color="secondary"
                >
                  <MenuIcon />
                </IconButton>
              </div>
            </Toolbar>
          </AppBars>
        </MuiThemeProvider>
      </div>
    );
  }
}
AppBar.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(AppBar);
