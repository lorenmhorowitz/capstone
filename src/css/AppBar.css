/*
#f96302 //orange home depot

*/

#root {
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
}
#grow {
  flex-grow: 1;
}

#logo {
  position: absolute;
  left: 20px;
}

#StoneTitle {
  color: #f96302;
  font-weight: bold;
  font-size: 33px;
}

#HomeTitle {
  color: #ffffff;
  font-weight: bold;
  font-size: 33px;
}
#menuButton {
  position: absolute;
  right: 20px;
  color: #ffffff;
}
