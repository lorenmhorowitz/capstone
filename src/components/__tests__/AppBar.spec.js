import React from "react";
import ReactDOM from "react-dom";
import AppBar from "../AppBar";
// import renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";

describe("AppBar Component", () => {
  const div = document.createElement("div");
  it("renders without crashing when no buttons are passed", () => {
    ReactDOM.render(<AppBar buttons={[]} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders without crashing when a button is passed", () => {
    const buttons = ["Logout"];
    let wrapper = mount(<AppBar buttons={buttons} />);
    expect(wrapper.exists()).toBe(true);
  });
  // it("accepts and handles a click on the logo", () => {
  //   const buttons = ["Logout"];
  //    let redirect = false;
  //   let wrapper = mount(<AppBar buttons={buttons}/>);

  //   expect(redirect).toBe(false);
  //   const toolbar = wrapper.find("logo");
  //   console.log(toolbar.html());

  //   toolbar.simulate("click", {onClick: () => {
  //     redirect= true
  //   }
  // });
  //   expect(redirect).toBe(true);
  // });
});
