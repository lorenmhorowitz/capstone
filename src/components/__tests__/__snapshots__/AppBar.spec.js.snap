// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Component renders correctly when a button and handler are passed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(AppBar)
    buttons={
      Array [
        "Logout",
      ]
    }
    handler={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "buttons": Array [
        "Logout",
      ],
      "classes": Object {},
      "handler": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "buttons": Array [
          "Logout",
        ],
        "classes": Object {},
        "handler": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`AppBar Component renders without crashing when a button is passed 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(AppBar)
    buttons={
      Array [
        "Logout",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "buttons": Array [
        "Logout",
      ],
      "classes": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "buttons": Array [
          "Logout",
        ],
        "classes": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
